spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulProxyConfiguration.java

```diff
@@ -88,7 +88,9 @@ public DiscoveryClientRouteLocator routeLocator() {
 	public PreDecorationFilter preDecorationFilter(RouteLocator routeLocator) {
 		return new PreDecorationFilter(routeLocator,
 				this.zuulProperties.isAddProxyHeaders(),
-				this.zuulProperties.isRemoveSemicolonContent());
+				this.zuulProperties.isRemoveSemicolonContent(),
+				this.server.getServletPrefix(),
+				this.zuulProperties);
```

Comment:
I struggled with that a bit.
Initially I wanted to externalize the "fallback" path also, so I needed 2 provide 2 more parameters to PreDecorationFilter: the configured fallback path + Zuul.servletPath.
The constructor signature would look ugly, so I decided to go for passing the ZuulProperties.
But I see  your point especially from the testing perspective, where just clear intended params without the generic ZuulProperties would keep things nicer.
On the long run though I think passing ZuulProperties directly will be the way to go in order to keep the constructor signature under control and do not rely on setters.
Maybe a fluent builder interface will help.

Do you think it would be good to just pass the servletPath as a parameter for now and do not pass ZuulProperties and we will see how to cintinue in the future?
There are alot